src/codegen/mod.rs

```diff
@@ -772,8 +772,26 @@ impl CodeGenerator for Type {
                     return;
```

Comment:
So there's this other piece of code that deals with enums... I guess that means that `NewType` doesn't work for enums at all... Should it?

Reply:
Would something like this be acceptable?
```rust
enum Foo {
    Bar,
    Baz,
}

#[repr(transparent)]
struct Meow(Foo);

impl Meow {
    pub const Bar: Meow = Meow(Foo::Bar);
    pub const Baz: Meow = Meow(Foo::Baz);
}
```

Reply:
I would make it just fall through the general code path, no need to generate the variants on top, IMO.